OSPSuite.Presentation.Importer/Core/IColumnDescription.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace OSPSuite.Presentation.Importer.Core
+{
+   public interface IColumnDescription
+   {
+      int Index { get; set; } //the unparsed name
+
+      IList<string> ExistingValues { get; set; }
```

Comment:
why is this an IList?

Reply:
Not sure if you mean it should be an IEnumerable (it is not because we are manipulationg the data) or a List.

Reply:
No. I mean why is this not an IReadOnlyList. ColumnDescription should be an immutable object in my opinion. The list of existingvalues should not change over time

Reply:
ColumnDescription is an entity just holding the values. The ExistingValues are manipulated from the caller. Sure we could add some methods for adding values to existing and then making the filed immutable from the outside but it will not change much I believe since you will still have an Add method right. I am not sure if we are removing from the outside, if we are not we could at least restrict this but again, from its usage I believe it is ok to have it as a list you could manipulate. Encapsulating all the logic for the columns in the ColumnDescription would mean we need to parse the whole data column and then pass the data to the class to fill the values. I am not sure it will be better but this way we could make sure the data is not corrupted afterwards. We could discuss more in our meetings.